src/libslic3r/PrintObject.cpp

```diff
@@ -2396,7 +2396,10 @@ void PrintObject::discover_horizontal_shells()
                                     // is grown, and that little space is an internal solid shell so 
                                     // it triggers this too_narrow logic.)
                                     internal));
-                            solid = new_internal_solid;
+
```

Comment:
I see the issue. I am not sure which solution is better, whether yours, or just not updating the "solid". Your solution may add some unnecessary infill, just not updating the "solid" may lead to infills not being supported. 

This whole function is destined for removal one day anyway, it does not ensure that the perimeters get supported, it is not parallelized.

Reply:
The 2.2.0-alpha4 has the line commented out.